### PR TITLE
Fixed background color to allow linear-gradient values

### PR DESCRIPTION
**Pull request by alansouzati**
#### What does this PR do?

Allows `linear-gradient(...)` to be used as a background color

#### Where should the reviewer start?

box.stories.js

#### What testing has been done on this PR?

storybook

#### How should this be manually tested?

open Custom Color inside the Box examples

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yup

#### Is this change backwards compatible or is it a breaking change?

compatible

### CODE REVIEW

File: src/js/components/Layer/StyledLayer.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export const StyledOverlay = styled.div`
   left: 0px;
   right: 0px;
   bottom: 0px;
-  background-color: ${props =>
+  background: ${props =>
     props.theme.layer.overlayBackgroundColor};
```
Comment (ericsoderberghp) [inline src/js/components/Layer/StyledLayer.js]: I think this needs a merge.

File: src/js/utils/styles.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { palm, parseMetricToNum } from './mixins';
 import { colorForName, colorIsDark, getRGBA } from './colors';
 
 export const activeStyle = css`
-  background-color: ${props => props.theme.global.hover.backgroundColor};
+  background: ${props => props.theme.global.hover.backgroundColor};
```
Comment (ericsoderberghp) [inline src/js/utils/styles.js]: I'm wondering if we should change this to `global.hover.background`, like I did for Layer, to allow more flexibility and to be more consistent?

Comment (alansouzati) [reply src/js/utils/styles.js]: can we do this in a separate pr ?

File: src/js/components/Layer/StyledLayer.js (hunk 1 above)
Comment (alansouzati) [reply src/js/components/Layer/StyledLayer.js]: what do you mean ?

Comment (ericsoderberghp) [reply src/js/components/Layer/StyledLayer.js]: Oops. I was thinking of something else. My bad.

File: src/js/utils/styles.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/utils/styles.js]: Yes.

**Pull request merged**